libtransmission/peer-io.h

```diff
@@ -98,7 +98,7 @@ class tr_peerIo final : public std::enable_shared_from_this<tr_peerIo>
 
     [[nodiscard]] constexpr auto socketAddress() const noexcept
     {
-        return socket.socketAddress();
+        return std::make_pair(addr_, port_);
```

Comment:
This doesn't look like it fixes the underlying issue. If `tr_peerIo` is destroyed and leads to crash here, then the fact that your changes work could be attributed to pure luck due to accessing data at a different offset (note that `socket` member is closer to the beginning of the object than your new `addr_` and `port_` members).

Reply:
Feel free to offer a better fix. In the mean time, that rollback works very well:

- without the rollback, I got a crash on pause 20/20 times.
- with the rollback, I got a crash on pause 0/20 times.

Reply:
I can understand why you would think it fixes the issue empirically, but I don't see any logical explanation why it would in fact fix anything.

Reply:
That's what I wrote in the PR description.
I spent half of my Sunday on this, to pinpoint precisely what line was causing the crash.

Because the alternative to a minimal change like this one is to rollback the last 12 pull requests that were merged on Main, until we're back to stable code... ;)